File: app/src/main/java/com/jqorz/test/receiver/MyReceiver.java
package com.jqorz.test.receiver;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import com.jqorz.test.ToastUtil;

/**
 * 测试广播接收器用法
 */
public class MyReceiver extends BroadcastReceiver {
    private OnListener onListener;

    public void setOnListener(OnListener onListener) {
        this.onListener = onListener;
    }

    @Override
    public void onReceive(Context context, Intent intent) {
        if (intent.getAction() != null && intent.getAction().equals("A")) {
            ToastUtil.showToast(context, "收到");
            if (onListener != null) {
                onListener.onChanged();
            }
        }
    }

    public interface OnListener {
        void onChanged();
    }
}

File: app/src/main/java/com/jqorz/test/broadcast/MyReceiver.kt
package com.jqorz.test.broadcast

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.text.TextUtils

/**
 * 测试广播接收器用法
 */
class MyReceiver : BroadcastReceiver() {
    var onReceiveBroadcastListener: OnReceiveBroadcastListener? = null

    override fun onReceive(context: Context, intent: Intent) {
        if (TextUtils.equals(intent.action, SendBroadActivity.Companion.INTENT_ACTION)) {
            onReceiveBroadcastListener?.onChanged(intent.getStringExtra("data"))
        }
    }

    interface OnReceiveBroadcastListener {
        fun onChanged(text: String?)
    }
}